aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -218,14 +224,18 @@ public boolean isMetricsEnabled() {
      *
      * @param userMetaDataMap userMetaData map
      */
-    public void addUserMetaData(Map<String, String> userMetaDataMap) {
-        StringBuilder userMetadata = new StringBuilder(this.userMetaData);
+    public void updateUserMetaData(Map<String, String> userMetaDataMap) {
+        StringBuilder userMetadata = new StringBuilder("?SDK=Android&Version=" + VersionInfoUtils.getVersion());
         int baseLength = userMetadata.length();
 
-        if (userMetaDataMap != null) {
-            // Append each of the user-specified key-value pair to the user metadata for the connection
+        // Update the meta data map
+        for(Map.Entry<String, String> metaData : userMetaDataMap.entrySet()){
+            this.userMetaDataMap.put(metaData.getKey(), metaData.getValue());
+        }
 
-            for (Map.Entry<String, String> metaData : userMetaDataMap.entrySet()) {
+        if (this.userMetaDataMap != null) {
```

Comment:
This null check shouldn't matter. And if it was null, then the action should be to construct a new map.